lib/autoproj/ops/configuration.rb

```diff
@@ -227,17 +242,30 @@ def load_and_update_package_sets(root_pkg_set, only_local = false)
                 end
                 by_repository_id[repository_id] = [vcs, imported_from]
 
+                # Make sure the package set has been already checked out to
+                # retrieve the actual name of the package set
                 if !vcs.local?
                     update_remote_package_set(vcs, only_local)
-                    create_remote_set_user_dir(vcs)
+                    raw_local_dir = PackageSet.raw_local_dir_of(vcs)
+                    required_remotes_dirs << raw_local_dir
                 end
-
                 name = PackageSet.name_of(manifest, vcs)
+
+                names = by_name.collect { |k,v| k }
+                Autoproj.debug "Trying to load package_set: #{name} from definition #{repository_id}"
+                Autoproj.debug "Already loaded package_sets are: #{names}"
+
                 if already_loaded = by_name[name]
                     already_loaded_pkg_set, already_loaded_vcs = *already_loaded
                     if already_loaded_vcs != vcs
                         if imported_from
-                            Autoproj.warn "#{imported_from.name} auto-imports a package set from #{vcs}, but a package set with the same name (#{name}) has already been imported from #{already_loaded_vcs}, I am skipping this one"
+                            Autoproj.warn "redundant auto-import by #{imported_from.name} for package set '#{name}'."
+                            Autoproj.warn "    A package set with the same name (#{name}) has already been imported from"
+                            Autoproj.warn "        #{already_loaded_vcs}"
+                            Autoproj.warn "        with server configuration '#{repository_configuration_label(already_loaded_vcs)}'."
```

Comment:
The vcs object is used here already. Just going through options[:repository id] which looks to me like a more stable interface in the given case!

``` ruby
[1] pry(#<Autoproj::Ops::Configuration>)> p vcs
#<Autoproj::VCSDefinition:0x00000001bb8838 @options={:push_to=>"git@github.com:/rock-core/package_set.git", :retry_count=>10, :repository_id=>"github:/rock-core/package_set.git"}, @url="git://github.com/rock-core/package_set.git", @type="git", @raw=[[nil, {"github"=>"rock-core/package_set"}]]>
=> #<Autoproj::VCSDefinition:0x00000001bb8838
 @options={:push_to=>"git@github.com:/rock-core/package_set.git", :retry_count=>10, :repository_id=>"github:/rock-core/package_set.git"},
 @raw=[[nil, {"github"=>"rock-core/package_set"}]],
 @type="git",
 @url="git://github.com/rock-core/package_set.git">
[2] pry(#<Autoproj::Ops::Configuration>)>